op-node/rollup/derive/engine_queue.go

```diff
@@ -483,22 +489,32 @@ func (eq *EngineQueue) Reset(ctx context.Context, _ eth.L1BlockRef) error {
 			safe, safe.Time, l1Origin, l1Origin.Time))
 	}
 
-	pipelineNumber := l1Origin.Number - eq.cfg.ChannelTimeout
-	if l1Origin.Number < eq.cfg.ChannelTimeout {
-		pipelineNumber = 0
+	// Walk back L2 chain to find the L1 origin that is old enough to start buffering channel data from.
+	pipelineL2 := safe
+	for pipelineL2.Number > eq.cfg.Genesis.L2.Number && pipelineL2.L1Origin.Number > eq.cfg.Genesis.L1.Number && pipelineL2.L1Origin.Number+eq.cfg.ChannelTimeout > l1Origin.Number {
```

Comment:
I don't love how long this condition is, perhaps could use a for loop with a break and assign each part of the condition to a variable like this:

```go
for {
    isPipelineNumberGreater := pipelineL2.Number > eq.cfg.Genesis.L2.Number
    isPipelineOriginGreater := pipelineL2.L1Origin.Number > eq.cfg.Genesis.L1.Number
   // other condition
   if isPipelineNumberGreater && isPipelineOriginGreater {
      // do logic
    }
}
```